File: apps/admin_mobile/lib/application/use_cases/reservations/get_available_slots_use_case.dart
import '../../../domain/entities/reservation.dart';
import '../../../domain/repositories/reservation_repository.dart';

class GetAvailableSlotsUseCase {
  final ReservationRepository _repo;
  GetAvailableSlotsUseCase(this._repo);

  Future<List<AvailableSlot>> call(String date, int serviceId) =>
      _repo.getAvailableSlots(date, serviceId);
}

File: apps/admin-v2/src/application/use-cases/reservations/get-available-slots.use-case.ts
import type { ReservationRepository } from '@/domain/repositories/reservation.repository';

export class GetAvailableSlotsUseCase {
  constructor(private repo: ReservationRepository) {}

  execute(date: string, serviceId: string, durationMin?: number) {
    return this.repo.getAvailableSlots(date, serviceId, durationMin);
  }
}
